test/metabase/models/dashboard_test.clj

```diff
@@ -838,6 +839,18 @@
       (t2/delete! Dashboard :id dashboard-id)
       (is (nil? (t2/select-one 'ParameterCard :card_id card-id))))))
 
+(deftest do-not-update-parameter-card-if-it-doesn't-change-test
+  (testing "Do not update ParameterCard if updating a Dashboard doesn't change the parameters"
+    (mt/with-temp [:model/Card      {source-card-id :id} {}
+                   :model/Dashboard {dashboard-id :id} {:parameters [{:name       "Category Name"
+                                                                      :slug       "category_name"
+                                                                      :id         "_CATEGORY_NAME_"
+                                                                      :type       "category"
+                                                                      :values_source_type    "card"
+                                                                      :values_source_config {:card_id source-card-id}}]}]
+        (with-redefs [parameter-card/upsert-or-delete-from-parameters! (fn [& _] (throw (ex-info "Should not be called" {})))]
```

Comment:
Rather use `mt/with-dynamic-redefs`, which is thread-safe.

Reply:
ah nice! 2ab819ea80